pkg/daemon/daemon.go

```diff
@@ -149,19 +157,128 @@ func (d *daemon) GetIbSriovNetwork(networkID string) (string, *utils.IbSriovCniS
 	return networkName, ibCniSpec, nil
 }
 
+// Get Pod's IB network specified by name. Update netMap with Pod's networks.
+func getIbSriovPodNetwork(netName string, pw *PodWrapper, netMap map[types.UID][]*v1.NetworkSelectionElement) error {
+	var err error
+	networks, ok := netMap[pw.pod.UID]
+	if !ok {
+		networks, err = netAttUtils.ParsePodNetworkAnnotation(pw.pod)
+		if err != nil {
+			return fmt.Errorf("failed to read pod networkName annotations pod namespace %s name %s, with error: %v",
+				pw.pod.Namespace, pw.pod.Name, err)
+		}
+
+		netMap[pw.pod.UID] = networks
+	}
+	network, err := utils.GetPodNetwork(networks, netName)
+	if err != nil {
+		return fmt.Errorf("failed to get pod networkName spec %s with error: %v", netName, err)
+	}
+
+	pw.ibNetwork = network
```

Comment:
what ensures that the retrieved network is indeed infiniband network ?

Reply:
> There is a check before the call for this method daemon.go to check if it contains ib-sriov-cni. I agree it not clear. Maybe we should move the check to this method

Currently main flow do next:
* for every network
* if network not IB - continue
* for every pod of that network
* get annotation of that network from pod

and doing check in this method looks weird, especially if network, which before loop we check is IB, cannot became non-IB.
Better then leave comment or rename function.

Reply:
make sense.  so rename it. Also let the method return the network and the and assigen the pw.ibNetwork = getPodNetwork(...). It wired that the method is called getPod but is not returning network